tutorials/developers/dev_nn_cross_val/dev_nn_cross_val.tex

```diff
@@ -0,0 +1,641 @@
+        'P', nncv.get('P'));
+end
+
+value = cellfun(@(d, nn) NNRegressorMLP_Evaluator('TEMPLATE', nne_template, 'D', d, 'NN', nn), ... ¥\circled{9}\circlednote{9}{initializes all of the \code{NNRegressorMLP\_Evaluator}s.}¥
+    d_list, nn_list, 'UniformOutput', false);
+
+\end{lstlisting}
+
+\begin{lstlisting}[
+	label={cd:m:NNRegressorMLP_CrossValidation:props},
+	caption={
+		{\bf NNRegressorMLP\_CrossValidation element props.}
+		The \code{props} section of generator code for \fn{\_NNRegressorMLP\_CrossValidation.gen.m} defines the properties to be used in \fn{NNRegressorMLP\_CrossValidation}.
+	}
+]
+%% ¡props!
+
+%%% ¡prop!
+P (parameter, scalar) is the permutation number.¥\circled{1}\circlednote{1}{defines the number for permuation feature importance.}¥
+%%%% ¡default!
+1e+2
+%%%% ¡check_prop!
+check = value > 0 && value == round(value);
+
+%%% ¡prop!
+AV_CORR (result, rvector) provides the metric of the correlation of coefficients. ¥\circled{2}\fourcirclednotes{2}{3}{4}{5}{calculate the average metrics, including the correlation of coefficients, the coefficient of determination, the mean absolute error, the mean squared error, and the root mean squared error.}¥
+%%%% ¡calculate!
+e_list = nncv.get('EVALUATOR_LIST');
+
+value = cellfun(@(e) e.get('CORR'), ...
+    e_list, 'UniformOutput', false);
+
+if isempty(value)
+    value = [];
+else
+    value = mean(cell2mat(value), 1);
+end
+
+%%% ¡prop!
+AV_DET (result, rvector) provides the coefficient of determination, a measure showing how well the predictions are replicated by the model. ¥\circled{3}¥
+%%%% ¡calculate!
+e_list = nncv.get('EVALUATOR_LIST');
+
+value = cellfun(@(e) e.get('DET'), ...
+    e_list, 'UniformOutput', false);
+
+if isempty(value)
+    value = [];
+else
+    value = mean(cell2mat(value), 1);
+end
+
+%%% ¡prop!
+AV_MAE (result, rvector) provides the metric of the mean absolute error. ¥\circled{4}¥
+%%%% ¡calculate!
+e_list = nncv.get('EVALUATOR_LIST');
+
+value = cellfun(@(e) e.get('MAE'), ...
+    e_list, 'UniformOutput', false);
+
+if isempty(value)
+    value = [];
+else
+    value = mean(cell2mat(value), 1);
+end
+
+%%% ¡prop!
+AV_MSE (result, rvector) provides the metric of the mean squared error. ¥\circled{5}¥
+%%%% ¡calculate!
+e_list = nncv.get('EVALUATOR_LIST');
+
+value = cellfun(@(e) e.get('MSE'), ...
+    e_list, 'UniformOutput', false);
+
+if isempty(value)
+    value = [];
+else
+    value = mean(cell2mat(value), 1);
+end
+
+%%% ¡prop!
+AV_RMSE (result, rvector) provides the metric of the root mean squared error. 
+%%%% ¡calculate!
+e_list = nncv.get('EVALUATOR_LIST');
+
+value = cellfun(@(e) e.get('RMSE'), ...
+    e_list, 'UniformOutput', false);
+
+if isempty(value)
+    value = [];
+else
+    value = mean(cell2mat(value), 1);
+end
+
+%%% ¡prop!
+AV_FEATURE_IMPORTANCE (result, cell) averages the feature importances across k folds. ¥\circled{6}\circlednote{6}{calculate the average feature importance.}¥
+%%%% ¡calculate!
```

Comment:
Maybe explain a bit more how feature importance is calculated. Athough it is explained a bit more in dev_nn_reg and dev_nn_cla I think it can be useful to include here more information in case someone reads this tutorial directly without reading the other two.

Reply:
<img width="171" alt="image" src="https://github.com/giovannivolpe/BRAPH-2-Matlab-beta/assets/30530538/0bf1fb41-e534-4411-a69c-a68d3f73a637">